FWCore/Framework/src/EventProcessor.cc

```diff
@@ -2256,6 +2260,7 @@ namespace edm {
                 // If appropriate, this will also start the next run.
                 endRunAsync(streamRunStatus_[iStreamIndex], iTask);
               }
+              status->noMoreEventsInLumi();
```

Comment:
This line needs to be moved just above line 2247. That was the bug (new in this PR) causing the seg fault. If line 2220 executes before this on another stream, the atomic counter is decremented before being set and global begin lumi gets skipped. I'll push a fix with this change soon along with Matti's suggested changes.